src/Microsoft.Extensions.Options/IOptions.cs

```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Extensions.Options
     /// Used to retreive configured TOptions instances.
     /// </summary>
     /// <typeparam name="TOptions">The type of options being requested.</typeparam>
-    public interface IOptions<out TOptions> where TOptions : class, new()
+    public interface IOptions<TOptions> where TOptions : class, new()
```

Comment:
> So the behavior difference between `IOptions` and `IOptionsSnapshot` is that IOptions will never change, its a cached singleton that never gets updated.

So `IOptions` might have been a singleton, and a`TOptions` once returned will never be mutated (by the Options feature at least) but but a new `TOptions` instance with new values will be returned after the cache is invalidated. Is this correct? If yes, I agree using a method is more appropriate than the `Value` property. Not sure that determines that the method should always take a name.